src/App.tsx

```diff
@@ -15,36 +17,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+const getPreparedGoods = (
+  goods: string[],
+  sortField: string,
+  reverseGoods: boolean,
+) => {
+  const prepareGoods = [...goods];
+
+  if (sortField) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseGoods) {
+    prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+};
+
+export const App: FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseGoods, setReverseGoods] = useState(false);
+
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    sortField,
+    reverseGoods,
+  );
+
+  const handleResetSort = () => {
+    setSortField('');
+    setReverseGoods(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => setSortField(SortType.ALPHABET)}
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortType.ALPHABET,
+          })}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => setSortField(SortType.LENGTH)}
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortType.LENGTH,
+          })}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => setReverseGoods(!reverseGoods)}
+          type="button"
+          className={cn('button is-warning', { 'is-light': !reverseGoods })}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField || reverseGoods) && (
```

Comment:
Create a variable for `sortField || reverseGoods` to avoid combining different logical operators